src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -310,6 +323,40 @@
     <MakeDir Directories="$(ProjectSpecificConfDir)" />
   </Target>
 
+  <!-- **************************************************************************** -->
+  <!-- In case of Razor compilation the protobuf files generated during the core compilation step are already moved into a temporary folder.
+       This target moves the protobuf files generated during the razor core compile step into a .razor directory, and alsomoves the protobuf files
+       from the temporary directory back to their original place. This target also takes care about generating the ProjectInfo.xml file for the razor part of the compilation.
+  -->
+  <!-- **************************************************************************** -->
+  <Target Name="CopyProjectInfoFileInPresenceOfRazor" AfterTargets="WriteSonarQubeProjectData" Condition=" $(TemporaryProjectSpecificOutDir) != '' ">
+    <PropertyGroup>
+      <RazorProjectSpecificOutDir>$(ProjectSpecificOutDir).razor</RazorProjectSpecificOutDir>
+    </PropertyGroup>
+    <PropertyGroup>
+      <ProjectInfo>$(ProjectSpecificOutDir)\ProjectInfo.xml</ProjectInfo>
+      <RazorProjectInfo>$(RazorProjectSpecificOutDir)\ProjectInfo.xml</RazorProjectInfo>
+    </PropertyGroup>
+    <!-- Move Razor specific protobuf files to their own directory. -->
+    <MoveDirectory SourceDirectory="$(ProjectSpecificOutDir)"
+               DestinationDirectory="$(RazorProjectSpecificOutDir)" />
+    <!-- Replace the out direcotry in the ProjectInfo.xml for the Razor build to point to the directory with .razor suffix. -->
+    <WriteLinesToFile File="$(RazorProjectInfo)"
+                  Lines="$([System.IO.File]::ReadAllText($(RazorProjectInfo)).Replace('$(ProjectSpecificOutDir)','$(RazorProjectSpecificOutDir)'))"
+                  Overwrite="true"
+                  Encoding="UTF-8"/>
+    <!-- Remove empty ProjectSpecificOutDir to be able to move the content of the temp direcotry. -->
+    <RemoveDir Directories="$(ProjectSpecificOutDir)" />
+    <!-- Move the protobuf files from the core compilation back to their original directory. -->
+    <MoveDirectory SourceDirectory="$(TemporaryProjectSpecificOutDir)"
+                   DestinationDirectory="$(ProjectSpecificOutDir)" />
+    <!-- Create a ProjetInfo.xml for the core compilation results and point it to the directory without any suffix. -->
+    <WriteLinesToFile File="$(ProjectInfo)"
+                  Lines="$([System.IO.File]::ReadAllText($(RazorProjectInfo)).Replace('$(RazorProjectSpecificOutDir)','$(ProjectSpecificOutDir)'))"
```

Comment:
`ProjectSpecificOutDir` is replaced with `RazorProjectSpecificOutDir` above to replace `RazorProjectSpecificOutDir` with `ProjectSpecificOutDir` here? That doesn't make sense and is very fragile.

Reply:
This one wasn't addressed, I still see two replacements there.